agent/Equalizer.py

```diff
@@ -0,0 +1,269 @@
+                elif i == 3:
+                    if start <= card < end:
+                        hearts.append(card)
+
+        suits = [clubs, diamonds, spades, hearts]
+        return suits
+
+    def played_cards_in_trick(self,
+                              partial_state: HeartsState):
+        """ Get the currently played cards in the trick"""
+        # why
+        trick_cards = []
+        card_index = 0
+        for i in partial_state.values:
+            if 20 < i < 35:
+                trick_cards.append(card_index)
+            card_index = card_index + 1
+        return trick_cards
+
+    def get_low_card(self,
+                     suit_to_choose_from: list):
+        """In this function we want to want to pick the lowest card from this suit"""
+        suit_to_choose_from.sort()
+        choice = suit_to_choose_from[0]
+        return choice
+    def passing(self, partial_state: HeartsState):
+        hand = self.sort_suits(partial_state)
+        # We want to void out Clubs - Diamonds - Spades - Hearts
+        pirority = [ 0 , 1, 2 ,3 ]
+        # Reference to all the amount of cards in each suit
+        suit_map = {
+                "D":len(hand[1]),
+                "C":len(hand[0]),
+                "S":len(hand[2]),
+                "H":len(hand[3])
+        }
+
+        cards_to_pass = []
+        suit_counter = 0
+        # Follows the order of priority to void suits.
+        for suit in hand:
+            # Clubs first since they are the first suit played
+            if len(suit) > 0 :
+                suit.reverse()
+                for card in suit:
+                    # Avoid giving away the queen of spades
+                    if (suit_counter != 2) & (card != 36):
+                        cards_to_pass.append(card)
+            if len(cards_to_pass) > 3:          # There is enough cards to pass
+                break
+            suit_counter += 1
+            # rinse and repeat the first part but with other suits
+        return  cards_to_pass[0:3]
+
+
+    def get_highest_safe_card(self,
+                              suit_to_choose_from: list,
+                              partial_state: HeartsState):
+        """In this function we want to pick the safest high card from the suit we need to follow"""
+        # want to pick the safest high card from the suit
+
+        suit_to_choose_from.sort()
+        # get the cards already played in this trick
+        trick_cards = self.played_cards_in_trick(partial_state)
+        queen_played = False
+        for card in trick_cards:
+            if card == 36:
+                # if the queen has been played, we're going to want to choose the lowest card we can
+                queen_played = True
+        # if the queen has been played or our points are over 10, this
+        # agent is going to play the lowest card it can
+        if queen_played or partial_state.points[self.player_position] >= 10:
+            choice = suit_to_choose_from[0]
```

Comment:
You should probably double check this, but Partial_state.score seems to always be [0,0,0,0] and never gets updated (line 259-260).
To checks points call the method points() in the adj.
Is player_position the position of the agent in the points list? If so, then the player_position is off by one. 
Right now in play_hearts.py agent_1 is the Equalizer. So in the points list, the equalizer's points should be at index 0.
I think this issue comes from where player_position is calculated in get_action() (line 44-48). Since our encoded values for player cards is 1-4 which i is set to (line 45), player position is set to 1.
If this is correct then this should fix it.
`        points = self.own_adj.points(partial_state)
        if queen_played or points[self.player_position - 1] >= 10:`

Reply:
So your fix worked in that the points are now updated properly, however looking at these values in the debugger it looks like this still doesn't do what I intend for it.

Is there any consistent way to see the total amount of points that have been scored by the agent over the course of the entire game?

Reply:
I think that would be partial_state.score? Which gets updated every round I believe...

Reply:
I believe you're correct.  I've switched over to using score instead of points now.